prover/src/trace/trace_lde/default/mod.rs

```diff
@@ -63,10 +63,11 @@ where
         trace_info: &TraceInfo,
         main_trace: &ColMatrix<E::BaseField>,
         domain: &StarkDomain<E::BaseField>,
+        partition_size: usize,
     ) -> (Self, TracePolyTable<E>) {
```

Comment:
Something doesn't feel right about passing `partition_size` here and also separately in `set_aux_trace()`. Maybe we should have a separate struct - e.g., `PartitionOptions` which could look like so:
```rust
pub struct PartitionOptions {
    num_partitions: usize,
    min_partition_size: usize,
}
```
Then we could put this struct into a filed and use it both here and in `set_aux_trace()`.

We could put the `parition_size()` method onto this struct.

Reply:
Just to emphasize that the `partition_size` passed in here an in `set_aux_trace()` need not be the same

Reply:
Yes, understood - but I still think having `PartitionOptions` would be ab it cleaner:

1. We'd be able to get it from `ProofOptions` as a single unit.
2. It would contain the logic for computing partition size.
3. We won't need other pass it to `TraceLde::set_aux_trace()` (because we'd put it as a single new filed into the default trace LDE) or `Prover::new_trace_lde()` (because we can get it from options).

Reply:
Makes sense, I have now introduced `PartitionOption` struct.